docs/configuration.md

```diff
@@ -143,6 +143,39 @@ fork_management:
     network_id: ae_uat
 ```
 
+### Hardforks
+
+Hardforks allow the specification of consensuss protocol versions with their repsective heights for custom chains.
+Optionally, prefunded account and contract files can be speficied which will be effective at these heights.
+The files can be specified using the absolute path or a path relative to the home of the node in the directory data/aecore.
+
+```yaml
+chain:
+    persist: true
+    hard_forks:
+        "genesis":
+            "height": 0
+            "accounts_file": hf_dir/genesis/accounts.json
+            "contracts_file": hf_dir/genesis/contracts.json
+        "first":
+            "height": 1800
+            "accounts_file": hf_dir/first/accounts.json
+        "seconf":
+            "height": 1100
+```
+
+For hyperchains a contact file can also be specified alongside the other files.
+```yaml
+chain:
+    persist: true
+    hard_forks:
+        "genesis":
+            "height": 0
+            "accounts_file": hf_dir/genesis/accounts.json
+            "hc_contracts_file": hf_dir/genesis/contracts.json
```

Comment:
The contracts file is applied when a certain height is hit but Lima is the last time it was used. The hyperchain contracts are called from hyperchain specific parts of the code and the format returned is different. I thought it might be needed to apply both types of contracts.

Reply:
Option 1 sounds like a good idea. At the moment, the consensus contracts are only called when creating the genesis block but I remember it was supposed to be possible to change the consensus. This would make it possible to change the consensus but I am not sure if anymore work would be needed.

Reply:
The contracts are applied differently. The lima hard fork contract states "Note that there are no call objects, and no transactions recorded on the chain, and no fees or gas compensation, is given for this execution." The functionality is "to enable distributing funds to accounts that missed the final migration deadline". At the moment, I check at every protocol change if there is a contracts file and if so apply the functionality above that was added with Lima. I could either not apply the contracts file after Lima or try to distinguish between the consensus and account migration contracts. The account migration contracts contains no owner, I could use that fact, or have the format of the file specify what functionality the contract supports.

Reply:
I don’t think we need to overwrite that then - “the old contracts”